packages/UI/src/components/atoms/Tag/Tag.tsx

```diff
@@ -0,0 +1,445 @@
+    //   color: 'blue',
+    //   selected: true,
+    //   className: 'hover:bg-blue-700',
+    // },
+    // {
+    //   color: 'orange',
+    //   selected: true,
+    //   className: 'hover:bg-orange-400',
+    // },
+
+    // setup outline variants for the colors
+    {
+      mode: 'dark',
+      outline: true,
+      className: 'border-white',
+    },
+
+    // selected colors variations with dark and light mode
+    {
+      color: 'white',
+      selected: true,
+      className: 'hover:bg-gray-200',
+    },
+    {
+      color: 'black',
+      selected: true,
+      className: 'hover:bg-gray-700',
+    },
+    {
+      color: 'cosmic',
+      selected: true,
+      mode: 'light',
+      className: 'hover:bg-brand-alt-cosmic-700',
+    },
+    {
+      color: 'cosmic',
+      selected: true,
+      mode: 'dark',
+      className: 'hover:bg-brand-alt-cosmic-500',
+    },
+    {
+      color: 'jupiter',
+      selected: true,
+      mode: 'light',
+      className: 'hover:bg-jupiter-700',
+    },
+    {
+      color: 'jupiter',
+      selected: true,
+      mode: 'dark',
+      className: 'hover:bg-jupiter-500',
+    },
+    {
+      color: 'nebula',
+      selected: true,
+      mode: 'light',
+      className: 'hover:bg-brand-alt-nebula-700',
+    },
+    {
+      color: 'nebula',
+      selected: true,
+      mode: 'dark',
+      className: 'hover:bg-brand-alt-nebula-500',
+    },
+    {
+      color: 'neptune',
+      selected: true,
+      mode: 'light',
+      className: 'hover:bg-neptune-700',
+    },
+    {
+      color: 'neptune',
+      selected: true,
+      mode: 'dark',
+      className: 'hover:bg-neptune-500',
+    },
+    {
+      color: 'saturn',
+      selected: true,
+      mode: 'light',
+      className: 'hover:bg-saturn-700',
+    },
+    {
+      color: 'saturn',
+      selected: true,
+      mode: 'dark',
+      className: 'hover:bg-saturn-500',
+    },
+    {
+      color: 'uranus',
+      selected: true,
+      mode: 'light',
+      className: 'hover:bg-uranus-700',
+    },
+    {
+      color: 'uranus',
+      selected: true,
+      mode: 'dark',
+      className: 'hover:bg-uranus-500',
+    },
+
+    // setup dark mode variations for the colors
+    {
+      color: 'cosmic',
+      mode: 'dark',
+      className: 'text-black bg-brand-alt-cosmic-300',
+    },
+    {
+      color: 'jupiter',
+      mode: 'dark',
+      className: 'bg-jupiter-300',
+    },
+    {
+      color: 'nebula',
+      mode: 'dark',
+      className: 'text-black bg-brand-alt-nebula-300',
+    },
+    {
+      color: 'neptune',
+      mode: 'dark',
+      className: 'text-black bg-neptune-300',
+    },
+    {
+      color: 'saturn',
+      mode: 'dark',
+      className: 'bg-saturn-300',
+    },
+    {
+      color: 'uranus',
+      mode: 'dark',
+      className: 'bg-uranus-300',
+    },
+  ],
+  defaultVariants: {
+    size: 'md',
+    color: 'black',
+    selected: false,
+    shadow: false,
+    outline: false,
+    mode: 'light',
+  },
```

Comment:
let's use the defaults directly in the component function scope as well

example:
```jsx
export function Tag({ size = 'md', color = 'black', selected = false, outline = false, mode: 'light' , ...props}: TagProps) 
```

Reply:
I used the single var as it would be duplicating the default variant definitions.  Then we still need to check for the default options as any parameter passed in would overwrite all those anyways.  By doing that, we would also have to restructure the props value to pass into the tailwind variants function.

Can definitely add it if you think it would be more readable.

Would look something like this:
`export function Tag({
  label,
  size = 'md',
  color = 'black',
  selected = false,
  shadow = false,
  outline = false,
  mode = 'light',
  ...options
}: TagProps) {
  // set the default properties
  const props = {
    ...tag.defaultVariants,
    ...options,
    label,
    size,
    color,
    selected,
    shadow,
    outline,
    mode,
  };`

Reply:
See updates for the component